src/Clip.cpp

```diff
@@ -245,193 +250,199 @@ Clip::~Clip()
 /// Set the current reader
 void Clip::Reader(ReaderBase* new_reader)
 {
-	// set reader pointer
 	reader = new_reader;
-
-	// set parent
-	reader->ParentClip(this);
-
-	// Init reader info struct and cache size
-	init_reader_settings();
+	if (reader) {
+		reader->ParentClip(this);
+		init_reader_settings();
+	}
 }
 
 /// Get the current reader
 ReaderBase* Clip::Reader()
 {
-	if (reader)
+	if (reader) {
 		return reader;
-	else
-		// Throw error if reader not initialized
-		throw ReaderClosed("No Reader has been initialized for this Clip.  Call Reader(*reader) before calling this method.");
+	} else {
+		throw ReaderClosed(
+			"No Reader has been initialized for this Clip. "
+			"Call Reader(*reader) before calling this method.");
+	}
 }
 
 // Open the internal reader
 void Clip::Open()
 {
-	if (reader)
-	{
-		// Open the reader
-		reader->Open();
-		is_open = true;
+	if (!reader) {
+		throw ReaderClosed(
+			"No Reader has been initialized for this Clip. "
+			"Call Reader(*reader) before calling this method.");
+	}
+	// Open the reader
+	reader->Open();
+	is_open = true;
 
-		// Copy Reader info to Clip
-		info = reader->info;
+	// Copy Reader info to Clip
+	info = reader->info;
 
-		// Set some clip properties from the file reader
-		if (end == 0.0)
-			End(reader->info.duration);
-	}
-	else
-		// Throw error if reader not initialized
-		throw ReaderClosed("No Reader has been initialized for this Clip.  Call Reader(*reader) before calling this method.");
+	// Set some clip properties from the file reader
+	if (end == 0.0)
+		End(reader->info.duration);
 }
 
 // Close the internal reader
 void Clip::Close()
 {
 	is_open = false;
-	if (reader) {
-		ZmqLogger::Instance()->AppendDebugMethod("Clip::Close");
 
-		// Close the reader
-		reader->Close();
+	if (!reader) {
+		throw ReaderClosed(
+			"No Reader has been initialized for this Clip. "
+			"Call Reader(*reader) before calling this method.");
 	}
-	else
-		// Throw error if reader not initialized
-		throw ReaderClosed("No Reader has been initialized for this Clip.  Call Reader(*reader) before calling this method.");
+
+	ZmqLogger::Instance()->AppendDebugMethod("Clip::Close");
+	reader->Close();
 }
 
 // Get end position of clip (trim end of video), which can be affected by the time curve.
 float Clip::End() const
 {
-	// if a time curve is present, use its length
-	if (time.GetCount() > 1)
-	{
-		// Determine the FPS fo this clip
-		float fps = 24.0;
-		if (reader)
-			// file reader
-			fps = reader->info.fps.ToFloat();
-		else
-			// Throw error if reader not initialized
-			throw ReaderClosed("No Reader has been initialized for this Clip.  Call Reader(*reader) before calling this method.");
-
-		return float(time.GetLength()) / fps;
+	if (!reader) {
+		// Throw error if reader not initialized
```

Comment:
```suggestion
```

Most of these offensively-redundant comments, I took out if I was touching the code around them, but I left this one in for some reason. Does someone think we can't _read_ the text directly below it!?